Zyphra Mamba2 and Flash Attention backward kernel blog (#25)
ROCm/rocm-blogs: blogs/ecosystems-and-partners/zyphra/README.md

---
blogpost: true
date: 10 December 2024

> blog_title: "Zyphra Introduces Frontier Training Kernels for Transformers and SSMs on AMD Instinct MI300X Accelerators"

Reviewer: Please use the same (simpler to understand...) title we already use in the blog's body:
```suggestion
blog_title: "Training Transformers and Hybrid models on AMD Instinct MI300X Accelerators"
```

Reviewer 2: Discussed over email earlier today, but we prefer to keep the original title

> blog_title: "Zyphra Introduces Frontier Training Kernels for Transformers and SSMs on AMD Instinct MI300X Accelerators"

Reviewer: Reverting back to an earlier suggested title, please use:
```suggestion
blog_title: "Zyphra Speeding Up Training on AMD Instinct MI300X Accelerators"
```

Reviewer 2: We prefer the previous title because:
1. We want readers to know we've achieved state of the art ("frontier")
2. We wrote kernels for both transformers and SSMs
3. We understand and resolved the main blocker from ROCm training: backward kernels

Reviewer: No worries, let's work with your preferred title!

thumbnail: 'amd_zyphra.jpg'
author: Quentin Anthony
tags: LLM, AI/ML, GenAI, PyTorch
category: Ecosystems and Partners
language: English
myst:
  html_meta:

>     "description lang=en": "In this blog, we demonstrate the first backwards kernels to surpass H100s for both transformers (Flash Attention v2) and hybrid models (Mamba2), which enables training foundation models on AMD Instinct MI300X accelerators."

Reviewer: This snippet is capped at 150 characters, I would like this to start with Zyphra & AMD work, with the message of suppressing others taking the back seat. Please see my revision: 
```suggestion
    "description lang=en": "This blog presents the training of Zyphra's backwards kernels for transformers and hybrid models on AMD Instinct MI300X accelerators, suppressing the H100s performance" 
```

Reviewer 2: resolved by https://github.com/ROCm/rocm-blogs/pull/25/commits/f069a059a08c72fed96f5224b6a220b5bf8ee322

    "keywords": "Mamba, PyTorch, S4, S6, Mamba2, Transformer, Flash Attention, Optimization, Hardware-aware, Transformer, Attention, ROCm, Mi210, MI250, MI300, AI/ML, Generative AI"
    "property=og:locale": "en_US"
---

# Training Transformers and Hybrid models on AMD Instinct MI300X Accelerators

> ## Harnessing the MI300 Hardware Specs

Reviewer: No section title here, please. The post just starts with the brief intro text, with no section title (this section title goes in, but only a little further, please see below). Thanks!
```suggestion
```

> # Training Transformers and Hybrid models on AMD Instinct MI300X Accelerators

Reviewer: Reverting back to an earlier suggested title, please use:
```suggestion
# Zyphra Speeding Up Training on AMD Instinct MI300X Accelerators
```

Reviewer 2: We prefer the previous title for this one as well

> ## Introduction

Reviewer: Please remove the "Introduction" section title. We follow a magazine-like (non-academic) approach in our blogs where each post starts with a brief introductory text, not explicitly titled Introduction.  ```suggestion
```

Reviewer 2: resolved by https://github.com/ROCm/rocm-blogs/pull/25/commits/f069a059a08c72fed96f5224b6a220b5bf8ee322

This blog is contributed by [Zyphra](https://www.zyphra.com/): a Palo Alto-based AI research lab and AMD Instinct Partner.

Zyphra is designing MaiaOS, a multimodal agent system that combines next-gen neural network architectures (SSM hybrids), long-term memory, and reinforcement learning.

In this blog we motivate our vision of training transformers and hybrid models at a lower cost using AMD technology. We explain how Zyphra harnessed the hardware advantages of the MI300x hardware for training both dense transformers and Zyphra's hybrid models. Specifically, the model blocks of interest are Mamba2 and Flash Attention v2. We conclude the blog by sharing benchmarks results showing the speedups we achieved on the MI300X using ROCm, compared to the competition.

> 

Reviewer: Please add a new section title here
```suggestion
## Harnessing the MI300 Hardware Specs
```

Reviewer 2: Resolved by https://github.com/ROCm/rocm-blogs/pull/25/commits/f069a059a08c72fed96f5224b6a220b5bf8ee322

> On paper, the AMD Instinct MI300X GPU accelerators contain some of the best hardware specifications on the market. The key hardware specs where the MI300X surpasses its main competitor, the NVIDIA H100 GPU, are High Bandwidth Memory (HBM) capacity and bandwidth.

Reviewer: Section title goes here, please:
```suggestion
## Harnessing the MI300 Hardware Specs

On paper, the AMD Instinct MI300X GPU accelerators contain some of the best hardware specifications on the market. The key hardware specs where the MI300X surpasses its main competitor, the NVIDIA H100 GPU, are High Bandwidth Memory (HBM) capacity and bandwidth.
```
